src/inmanta/server/services/resourceservice.py

```diff
@@ -222,14 +222,15 @@ async def get_resource(
         status: bool,
         log_action: const.ResourceAction,
         log_limit: int,
+        connection: Optional[Connection] = None,
     ) -> Apireturn:
         # Validate resource version id
         try:
             Id.parse_resource_version_id(resource_id)
         except ValueError:
             return 400, {"message": f"{resource_id} is not a valid resource version id"}
 
-        resv = await data.Resource.get(env.id, resource_id)
+        resv = await data.Resource.get(env.id, resource_id, connection)
```

Comment:
If `connection is None`, this method will acquire a connection twice. Once inside the `data.Resource.get()` method and once inside the `data.ResourceAction.get_log()`. We should make sure that a connection is only acquired once and reused for both queries.